modules/effects/arrow-effects/src/main/kotlin/arrow/effects/internal/IOPlatform.kt

```diff
@@ -0,0 +1,16 @@
+package arrow.effects.internal
+
+internal object IOPlatform {
+
+  /**
+   * Composes multiple errors together, meant for those cases in which error suppression, due to a second error being
+   * triggered, is not acceptable.
+   *
+   * On top of the JVM this function uses Throwable#addSuppressed, available since Java 7. On top of JavaScript the
+   * function would return a CompositeException.
+   */
+  fun composeErrors(first: Throwable, rest: List<Throwable>): Throwable {
```

Comment:
Used from `CancelUtils` which is used from `IOConnection`. Not sure if we want to get into this rabbit hole though.